File: py/1-two-sum.py
from typing import List


def twoSumBruteForce(nums: List[int], target: int) -> List[int]:  # O(n^2) time

    SIZE = len(nums)
    indices = []

    for i in range(0, SIZE - 1):
        for j in range(i + 1, SIZE):
            if (nums[i] + nums[j] == target):
                indices.append(i)
                indices.append(j)

        return indices


def twoSumOnePassHashtable(nums: List[int], target: int) -> List[int]:
    indices = [None] * 2
    numsHashtable = {}

    for i in range(len(nums)):
        complement = target - nums[i]
        if(complement in numsHashtable):
            indices = [numsHashtable[complement], i]
            return indices

        numsHashtable[nums[i]] = i

    return None

    # test
nums = [2, 4, 4, 7]
target = 8
print(twoSumOnePassHashtable(nums, target))

File: ts/1-two-sum.ts
// Time: 0(n^2)
function twoSumBruteForce(nums: number[], target: number): number[] {
        const size = nums.length;
        let indices: number[] = new Array(2);

        for (var i = 0; i < size - 1; i++) {
                for (var j = i + 1; j < size; j++) {
                        if (nums[i] + nums[j] == target) {
                                indices = [i, j];
                        }
                }
        }

        return indices;
};

// test
var nums = [2, 4, 8];
var target = 7;
console.log(twoSumBruteForce(nums, target));
